build/do.rq

```diff
@@ -27,6 +27,17 @@ package script
 
 import future.keywords
 
+# Path to Regal's project root. (./build/do.rq/../.. => ./)
+regal_root := rq.abs(rq.dir(rq.dir(rq.scriptpath())))
+
+# Root of the directory hierarchy where engine-specific capabilities JSON for
+# embedding into the Regal binary should be placed.
+embedded_caps_dir := rq.joinpath([regal_root, "pkg", "capabilities", "embedded"])
+
+# Directory where EOPA-specific JSON capabilities for embedding should be
+# placed.
+eopa_caps_dir := rq.joinpath([embedded_caps_dir, "eopa"])
+
```

Comment:
Hah. Eating your own dog food 🙃

Reply:
I did get to try out a few patterns that I had in mind when I implemented various features/builtins, and I think they have held up well under real usage.

I ended up not needing it in the final version, but I did get to use `rq.template()`'s quoting feature. It's nice to be confident that strings being subbed into shell commands will have correct quoting, without having to fuss with backslashes and stuff. 

e.g. 

```plain
$ rq --raw 'rq.template("./myscript.sh {{.path | quote}}", {"path": "/some/path with/spaces in it"})'
./myscript.sh "/some/path with/spaces in it"
```

I think I still need to see a use case for this that does not involve shell scripting before I advocate for upstreaming rq's extras into OPA's template builtin though.